Content.Server/Fax/FaxSystem.cs

```diff
@@ -325,10 +345,36 @@ private void UpdateAppearance(EntityUid uid, FaxMachineComponent? component = nu
         if (!Resolve(uid, ref component))
             return;
 
-        if (component.InsertingTimeRemaining > 0)
-            _appearanceSystem.SetData(uid, FaxMachineVisuals.VisualState, FaxMachineVisualState.Inserting);
-        else if (component.PrintingTimeRemaining > 0)
+        if (component.InsertingTimeRemaining > 0 && component.PaperSlot.Item != null)
+        {
+
+            if (HasComp<MobStateComponent>(component.PaperSlot.Item))
+            {
+
+                if (_tagSystem.HasTag(component.PaperSlot.Item.Value, "Hamster"))
+                {
+                    _appearanceSystem.SetData(uid, FaxMachineVisuals.VisualState, FaxMachineVisualState.InsertingHamlet);
+                }
+
+                if (_tagSystem.HasTag(component.PaperSlot.Item.Value, "MothRoach"))
+                {
+                    _appearanceSystem.SetData(uid, FaxMachineVisuals.VisualState, FaxMachineVisualState.InsertingMothroach);
+
+                }
+
+                if (_tagSystem.HasTag(component.PaperSlot.Item.Value, "Mouse"))
+                {
+                    _appearanceSystem.SetData(uid, FaxMachineVisuals.VisualState, FaxMachineVisualState.InsertingMouse);
+                }
+            }
+            else
+                _appearanceSystem.SetData(uid, FaxMachineVisuals.VisualState, FaxMachineVisualState.Inserting);
+        }
```

Comment:
Just have the data set the state directly instead of mapping the enums to a big if-else statement for tags.

Reply:
im now using a switch with a string to determine state. Hope this is acceptable.